backend/src/services/notifications.ts

```diff
@@ -0,0 +1,132 @@
+import * as admin from "firebase-admin";
+import schedule from "node-schedule";
+import { Mentee, Mentor, Session } from "../models";
+import { InternalError } from "../errors";
+
+/**
+ * Unicode notes:
+ * eyes - 1F440
+ * cool with glasses - 1F60E
+ * check mark - 2705
+ * broken heart - 1F494
+ *
+ * To find emojis and their unicode value, use https://apps.timwhitlock.info/emoji/tables/unicode
+ * The unicode is in format "U+1234"; in code, use "\u{1234}"
+ */
+
+async function sendNotification(title: string, body: string, deviceToken: string) {
+  const message = {
+    notification: {
+      title,
+      body,
+    },
+    token: deviceToken,
+  };
+
+  admin
+    .messaging()
+    .send(message)
+    .then((response) => {
+      console.log("Notification sent successfully:", response);
+    })
+    .catch((error) => {
+      console.error("Error sending notification:", error);
+    });
+}
+
+async function startUpcomingSessionCronJob() {
+  schedule.scheduleJob("*/1 * * * *", async () => {
+    try {
+      const upcomingNotifSessions = await Session.find({
+        upcomingSessionNotifSent: { $eq: false },
+      });
+      upcomingNotifSessions.forEach(async (session) => {
+        const dateNow = new Date();
+        const mentee = await Mentee.findById(session.menteeId);
+        if (!mentee) {
+          throw InternalError.ERROR_GETTING_MENTEE;
```

Comment:
Not sure if throwing an error would be the best way to handle errors in this scenario. Suppose that we have 10 sessions and only the 5th one is corrupted. From my understanding the function will get to the 5th session but throw an error. However, it won't continue to the session and send the proper notification for that session. I think its best for the function to just continue to the next session rather than ending the loop altogether.